src/Compilers/CSharp/Portable/Symbols/Extensions/RewrittenParameterSymbol.cs

```diff
@@ -22,8 +24,31 @@ public RewrittenParameterSymbol(ParameterSymbol originalParameter) :
 
         internal sealed override int CallerArgumentExpressionParameterIndex => _underlyingParameter.CallerArgumentExpressionParameterIndex;
 
-        internal sealed override ImmutableArray<int> InterpolatedStringHandlerArgumentIndexes => throw ExceptionUtilities.Unreachable(); // Tracked by https://github.com/dotnet/roslyn/issues/76130 : Follow up
+        internal sealed override ImmutableArray<int> InterpolatedStringHandlerArgumentIndexes
+        {
+            get
+            {
+                var originalIndexes = this._underlyingParameter.InterpolatedStringHandlerArgumentIndexes;
+                if (originalIndexes.IsDefaultOrEmpty)
+                {
+                    return originalIndexes;
+                }
+
+                // If this is the extension method receiver (ie, parameter 0), then any non-empty list of indexes must
+                // be an error, and we should have already returned an empty list.
+                Debug.Assert(_underlyingParameter.ContainingSymbol is not NamedTypeSymbol);
+                Debug.Assert(originalIndexes.All(static index => index != BoundInterpolatedStringArgumentPlaceholder.InstanceParameter));
+                return originalIndexes.SelectAsArray(static (index) => index switch
+                {
+                    BoundInterpolatedStringArgumentPlaceholder.InstanceParameter => throw ExceptionUtilities.Unreachable(),
+                    BoundInterpolatedStringArgumentPlaceholder.ExtensionReceiver => 0,
+                    BoundInterpolatedStringArgumentPlaceholder.TrailingConstructorValidityParameter => BoundInterpolatedStringArgumentPlaceholder.TrailingConstructorValidityParameter,
+                    BoundInterpolatedStringArgumentPlaceholder.UnspecifiedParameter => BoundInterpolatedStringArgumentPlaceholder.UnspecifiedParameter,
```

Comment:
Should we instead have case `var i when i < 0 => i` to handle other possible values in the future? Otherwise consider asserting that `index >= 0` in the case where we return `index + 1`.